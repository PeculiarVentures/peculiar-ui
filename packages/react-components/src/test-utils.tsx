import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
