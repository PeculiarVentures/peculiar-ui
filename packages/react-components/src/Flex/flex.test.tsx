import { renderWithWrapper as render } from '../test-utils';
import { Flex } from './index';

describe('<Flex />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <Flex>1</Flex>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render size auto', () => {
    const { asFragment } = render(
      <Flex size="auto" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render size grow', () => {
    const { asFragment } = render(
      <Flex size="grow" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <Flex className="my-class-name" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass component', () => {
    const { asFragment } = render(
      <Flex component="p" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
