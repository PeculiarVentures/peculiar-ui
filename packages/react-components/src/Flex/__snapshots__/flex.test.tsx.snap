// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 0px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

<div
    class="my-class-name emotion-0"
  />
</DocumentFragment>
`;

exports[`<Flex /> should pass component 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 0px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<Flex /> should render size auto 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  max-width: none;
  width: auto;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<Flex /> should render size grow 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 0px;
  -webkit-flex-basis: 0px;
  -ms-flex-preferred-size: 0px;
  flex-basis: 0px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`<Flex /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 0px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

<div
    class="emotion-0"
  >
    1
  </div>
</DocumentFragment>
`;
